couchdb/source/cmd/webhook/main.go

```diff
@@ -0,0 +1,169 @@
+	kubeconfig = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+)
+
+func getRegistrationDelayTime(rdt string) time.Duration {
+	var RegistrationDelay time.Duration
+
+	if rdt != "" {
+		rdtime, err := strconv.ParseInt(rdt, 10, 64)
+		if err != nil {
+			log.Fatalf("Error ParseInt: %v", err)
+		}
+
+		RegistrationDelay = time.Duration(rdtime)
+	}
+
+	return RegistrationDelay
+}
+
+func main() {
+	flag.Parse()
+
+	// Set up signals so we handle the first shutdown signal gracefully.
+	ctx := signals.NewContext()
+
+	// Report stats on Go memory usage every 30 seconds.
+	msp := metrics.NewMemStatsAll()
+	msp.Start(ctx, 30*time.Second)
+	if err := view.Register(msp.DefaultViews()...); err != nil {
+		log.Fatalf("Error exporting go memstats view: %v", err)
+	}
+
+	cfg, err := sharedmain.GetConfig(*masterURL, *kubeconfig)
+	if err != nil {
+		log.Fatal("Failed to get cluster config:", err)
+	}
+
+	log.Printf("Registering %d clients", len(injection.Default.GetClients()))
+	log.Printf("Registering %d informer factories", len(injection.Default.GetInformerFactories()))
+	log.Printf("Registering %d informers", len(injection.Default.GetInformers()))
+
+	ctx, _ = injection.Default.SetupInformers(ctx, cfg)
+	kubeClient := kubeclient.Get(ctx)
+
+	config, err := sharedmain.GetLoggingConfig(ctx)
+	if err != nil {
+		log.Fatal("Error loading/parsing logging configuration:", err)
+	}
+	logger, atomicLevel := logging.NewLoggerFromConfig(config, logconfig.WebhookName())
+	defer logger.Sync()
+	logger = logger.With(zap.String(logkey.ControllerType, logconfig.WebhookName()))
+
+	if err := version.CheckMinimumVersion(kubeClient.Discovery()); err != nil {
+		logger.Fatalw("Version check failed", zap.Error(err))
+	}
+
+	logger.Infow("Starting the Eventing Webhook")
+
+	// Watch the logging config map and dynamically update logging levels.
+	configMapWatcher := configmap.NewInformedWatcher(kubeClient, system.Namespace())
+	// Watch the observability config map and dynamically update metrics exporter.
+	configMapWatcher.Watch(metrics.ConfigMapName(), metrics.UpdateExporterFromConfigMap(logconfig.WebhookName(), logger))
+	// Watch the observability config map and dynamically update request logs.
+	configMapWatcher.Watch(logging.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, logconfig.WebhookName()))
+
+	if err = configMapWatcher.Start(ctx.Done()); err != nil {
+		logger.Fatalw("Failed to start the ConfigMap watcher", zap.Error(err))
+	}
+
+	var env envConfig
+	if err := envconfig.Process("", &env); err != nil {
+		log.Fatal("Failed to process env var", zap.Error(err))
+	}
+	registrationDelay := getRegistrationDelayTime(env.RegistrationDelayTime)
+
+	stats, err := webhook.NewStatsReporter()
+	if err != nil {
+		logger.Fatalw("failed to initialize the stats reporter", zap.Error(err))
+	}
+
```

Comment:
this part was bugging me before: let's see if we get an issue to share this better.  similar/same code is in different "core eventing" parts as well

Reply:
e.g. like `sharedmain.Main` ?